### PR TITLE
Factor enum classes Basis and Quadrature into their own files

### PR DESCRIPTION
**Pull request by kidder**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/NumericalAlgorithms/Spectral/Basis.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+#include "Options/Options.hpp"
+#include "Options/ParseOptions.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/GetOutput.hpp"
+#include "Utilities/StdHelpers.hpp"
+
+namespace Spectral {
+
+std::vector<Basis> all_bases() {
```
Comment (nilsvu) [inline src/NumericalAlgorithms/Spectral/Basis.cpp]: Switch to std::array? (Unrelated to this PR, just noticed while you're moving the code)

**Review by nilsvu — DISMISSED** [review]
I wonder if / how much this improved compile time. Do you have an idea @kidder? I can also do a test.

**Comment by kidder** [comment]
> I wonder if / how much this improved compile time. Do you have an idea @kidder? I can also do a test.

no

**Comment by nilsvu** [comment]
CI wants you to `include <vector>` in `Basis.hpp`

**Comment by nilsvu** [comment]
Timed a build before and after this commit and found no difference in build time. Still a good change.

File: src/NumericalAlgorithms/Spectral/Basis.hpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
+
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <iosfwd>
```
Comment (knelli2) [inline src/NumericalAlgorithms/Spectral/Basis.hpp]:
```cpp
#include <string>
#include <vector>
```

**Review by knelli2 — CHANGES_REQUESTED** [review]

**Comment by kidder** [comment]
@nilsvu @knelli2 made both changes, rebased, and squashed

**Review by nilsvu — DISMISSED** [review]

**Review by knelli2 — DISMISSED** [review]

File: src/NumericalAlgorithms/Spectral/Quadrature.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+#include <ostream>
+#include <string>
+
+#include "Options/Options.hpp"
+#include "Options/ParseOptions.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/GetOutput.hpp"
+#include "Utilities/StdHelpers.hpp"
+
+namespace {}  // namespace
```
Comment (wthrowe) [inline src/NumericalAlgorithms/Spectral/Quadrature.cpp]: ?

Comment (kidder) [reply src/NumericalAlgorithms/Spectral/Quadrature.cpp]: well there used to be something in the namespace...
fixed...

**Review by wthrowe — APPROVED** [review]

**Pull request merged**